addons/controllers/clusterbootstrap_controller.go

```diff
@@ -607,3 +624,64 @@ func (r *ClusterBootstrapReconciler) watchProvider(providerRef *corev1.TypedLoca
 		},
 	)
 }
+
+func (r *ClusterBootstrapReconciler) getCNI(
+	clusterBootstrapTemplate *runtanzuv1alpha3.ClusterBootstrapTemplate,
+	cluster *clusterapiv1beta1.Cluster,
+	log logr.Logger) (*runtanzuv1alpha3.ClusterBootstrapPackage, error) {
+
+	var clusterBootstrapPackage *runtanzuv1alpha3.ClusterBootstrapPackage
+
+	selectedCNI, err := r.parseClusterVariableString(cluster, r.ClusterBootstrapConfig.CNISelectionClusterVariableName)
+	if err != nil {
+		log.Error(err, "Error parsing cluster variable value for the CNI selection")
+		return nil, err
+	}
+	if selectedCNI != "" {
+		for _, cni := range clusterBootstrapTemplate.Spec.CNIs {
+			if selectedCNI == packageShortName(cni.RefName) {
+				clusterBootstrapPackage = cni
+				break
+			}
+		}
+	} else {
+		if len(clusterBootstrapTemplate.Spec.CNIs) > 0 {
+			clusterBootstrapPackage = clusterBootstrapTemplate.Spec.CNIs[0]
+		} else {
+			return nil, errors.New("no CNI was specified in the ClusterClass or in ClusterBootstrap.Spec.CNIs")
+		}
+	}
+
+	return clusterBootstrapPackage, nil
+}
+
+func (r *ClusterBootstrapReconciler) parseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName string) (string, error) {
```

Comment:
Could this be a standalone helper in pkg/util? Nothing specific to ClusterBootstrapReconciler in the method. 

Suggestion: Move to pkg/util and write a couple of unit tests.